### DISCUSSION TITLE
Is non-idempotency considered a bug?

### DISCUSSION THREAD

**Discussion by kirkwaiblinger** [prettier/prettier#16120] [General]
I have two different examples I've encountered in the wild of non-idempotent output, and my instinct is to report them as a bug, but I cannot find any documentation to suggest that this is unintended behavior. That being said, the presence of the "show second format" option on the playground suggests to me that it may be considered problematic. Please advise! Whether or not non-idempotency is bug-report-worthy, can that get added to the docs to reduce confusion? Thanks in advance!

**Comment by kachkaev** [comment]
👋 @kirkwaiblinger! You are correct. If Prettier amends your code on the second format, it’s a bug. Please check existing issues labeled as https://github.com/prettier/prettier/labels/area%3Aidempotency. If you observe something new that has not been reported, feel free to submit a new issue.

**Reply by kirkwaiblinger** [reply]
Can we make action items (that I'd be happy to take on) to

1. Mention this somewhere in the docs. One idea could be to add a little "info" block on the basic usage guide with a friendly message that says 
   > If you run `prettier --write .` followed by `prettier --check .` and it doesn't pass, that's a bug! Please report it on our github (hyperlinked to github issues page) or by creating a reproduction in the playground (link to playground).  In general, multiple passes of `prettier --write .` should never be necessary. For those interested, the technical term for this is that prettier should always be "idempotent".

1. In the prettier playground, with the "show second format" option enabled, if there's a difference in the format, put up a notice that says 
   > Congrats! You've found a bug! The second format differs from the first for your input. **Please report this to us** by clicking "Report Issue" in the bottom right corner. 
   
   I've used the playground with the "second format" enabled to figure out what's going on with a file that takes multiple passes, but I never previously reported it as a bug, since it wasn't obvious to me that it's considered one.

**Reply by kirkwaiblinger** [reply]
In the meantime I did infer from seeing the "area:idempotency" label while perusing the issues that is non-idempotency was definitely considered a bug (which is why I had originally closed the issue) and did report the issues I found 🙂
